src/main/java/com/group12/LaunchInterceptorConditionCollection.java

```diff
@@ -137,6 +137,42 @@ public boolean LIC7(List<Point> points, int kPts, double length1) throws Illegal
         return false;
     }
 
+    /**
+     * LIC #10 checks if there exists at least one set of three data points separated by exactly <b>ePts</b> and <b>fPts</b>
+     * consecutive intervening points, respectively, that are the vertices of a triangle with area greater
+     * than area1. The condition is not met when <b>points</b> has less than 5 elements.
+     *
+     * @param points list of radar echos ({@link Point})
+     * @param ePts first consecutive intervening points
+     * @param fPts second consecutive intervening points
+     * @param area1 the area of the triangle
+     * @return true if at least one set of three data points separated by exactly <b>ePts</b> and <b>fPts</b>
+     * consecutive intervening points, respectively forms a triangle with an area greater than <b>area1</b>, false otherwise.
+     * @throws IllegalArgumentException is thrown if <b>points</b> is null,
+     *                                  if <b>ePts</b> or <b>fPts</b> is less than 1, or if the length
+     *                                  of <b>points</b> - 3 is less than <b>ePts</b> + <b>fPts</b>.
+     */
+    public boolean LIC10(List<Point> points, int ePts, int fPts, double area1) throws IllegalArgumentException {
+        if (points == null) {
+            throw new IllegalArgumentException("Points list cannot be null");
+        }
```

Comment:
Don't forget to add a check that checks if the number of points is less than 5. Because if it is, then the condition is not met (i.e. LIC10 should return `false`). This is in the specification.

![bild](https://user-images.githubusercontent.com/70259607/151665038-bbea42b9-24ad-4f0b-aec1-7ce4b3f306c8.png)

Reply:
yes, will fix.

Reply:
This is already automatically tested because of the other conditions. Also, when NUMPOINTS is 4 for example, the exception "The size of points - 3 must be more or equal to ePts + fPts" will first trigger (so it can't be tested separately).

Reply:
Where is it tested automatically, could you clarify? I might have missed something.

What you could do is, right after your `if (points == null)` check, have a 
`
if (points.size < 5) {
    return false;
}

`

Reply:
If numpoints < 5, then either ```points.size() - 3 < ePts + fPts``` or ```ePts < 1 || fPts < 1``` (so an exception will be called).

Reply:
You are correct, an exception is thrown. I think the function should return `false` and not throw an exception when numpoints < 5.

Reply:
That's discussible, maybe we can make an issue about that?

Reply:
(Now it returns false when < 5, just that it won't come to that point)

Reply:
Sure, we can discuss and clarify this on a meeting.

Meanwhile, if you add an if statement right after the `if (points == null)` check that returns false, then you will reach that point. But if you add it after `points.size() - 3 < ePts + fPts` then it will not be reached.